unittests/test_parameters.py

```diff
@@ -3,120 +3,109 @@
+
+import reframe as rfm
+
+
+class NoParams(rfm.RunOnlyRegressionTest):
+    pass
+
+
+class TwoParams(NoParams):
+    parameter('P0', 'a')
+    parameter('P1', 'b')
+
+
+class Abstract(TwoParams):
+    parameter('P0')
+
+
+class ExtendParams(TwoParams):
+    parameter('P1', 'c', 'd', 'e', inherit_params=True)
+    parameter('P2', 'f', 'g')
 
 
 def test_param_space_is_empty():
     class MyTest(NoParams):
         pass
 
-    assert MyTest._rfm_params == {}
+    assert MyTest.param_space.is_empty
 
 
 def test_params_are_present():
     class MyTest(TwoParams):
         pass
 
-    assert MyTest._rfm_params['P0'] == ('a',)
-    assert MyTest._rfm_params['P1'] == ('b',)
+    assert MyTest.param_space['P0'] == ('a',)
+    assert MyTest.param_space['P1'] == ('b',)
 
 
 def test_param_override():
     class MyTest(TwoParams):
         parameter('P1', '-')
 
-    assert MyTest._rfm_params['P0'] == ('a',)
-    assert MyTest._rfm_params['P1'] == ('-',)
+    assert MyTest.param_space['P0'] == ('a',)
+    assert MyTest.param_space['P1'] == ('-',)
 
 
 def test_param_inheritance():
     class MyTest(TwoParams):
         parameter('P1', 'c', inherit_params=True)
 
-    assert MyTest._rfm_params['P0'] == ('a',)
-    assert MyTest._rfm_params['P1'] == ('b', 'c',)
+    assert MyTest.param_space['P0'] == ('a',)
+    assert MyTest.param_space['P1'] == ('b', 'c',)
 
 
 def test_filter_params():
     class MyTest(ExtendParams):
         parameter('P1', inherit_params=True, filter_params=lambda x: x[2:])
 
-    assert MyTest._rfm_params['P0'] == ('a',)
-    assert MyTest._rfm_params['P1'] == ('d', 'e',)
-    assert MyTest._rfm_params['P2'] == ('f', 'g',)
+    assert MyTest.param_space['P0'] == ('a',)
+    assert MyTest.param_space['P1'] == ('d', 'e',)
+    assert MyTest.param_space['P2'] == ('f', 'g',)
 
 
 def test_is_abstract_test():
     class MyTest(Abstract):
         pass
 
-    assert MyTest.is_abstract_test()
-
-
-def test_param_len_is_zero():
-    class MyTest(Abstract):
-        pass
-
-    assert MyTest.param_space_len() == 0;
+    assert MyTest.is_abstract
 
 
-def test_extended_param_len():
-    class MyTest(ExtendParams):
+def test_is_not_abstract_test():
+    class MyTest(TwoParams):
         pass
 
-    assert MyTest.param_space_len() == 8
-
-
-def test_param_iterator_is_empty():
-    class MyTest(Abstract):
-        @classmethod
-        def check_param_iterator(cls):
-            try:
-                tmp = next(cls._rfm_param_space_iter)
-                return False
-
-            except StopIteration:
-                return True
-
-            except:
-                return False
-
-    MyTest.prepare_param_space()
-    assert MyTest.check_param_iterator()
+    assert not MyTest.is_abstract
 
 
-def test_params_are_none():
+def test_param_len_is_zero():
     class MyTest(Abstract):
         pass
 
-    test = MyTest()
-    assert test.P0 == None
-    assert test.P1 == None
+    assert len(MyTest.param_space) == 0;
```

Comment:
I think it's good if you also do `assert MyTest.param_space == ??`

Reply:
I've added a new test that does the `assert MyTest.param_space == ...` in a separate test. I think it's cleaner to test the build of the param space and the param initialisation in different tests.